plugins/inventory/stackpath_compute.py

```diff
@@ -0,0 +1,281 @@
+        except KeyError:
+            raise AnsibleError("config missing client_id, a required paramter")
+        try:
+            client_secret = config['client_secret']
+            if client_secret != 64:
+                raise AnsibleError("client_secret must be 64 characters long")
+        except KeyError:
+            raise AnsibleError("config missing client_id, a required paramter")
+        return True
+
+    def _set_credentials(self):
+        '''
+            :param config_data: contents of the inventory config file
+        '''
+        self.client_id = self.get_option('client_id')
+        self.client_secret = self.get_option('client_secret')
+
+    def _authenticate(self):
+        payload = json.dumps(
+            {
+                "client_id": self.client_id,
+                "client_secret": self.client_secret,
+                "grant_type": "client_credentials",
+            }
+        )
+        headers = {
+            "Content-Type": "application/json",
+        }
+        resp = open_url(
+            self.api_host + '/identity/v1/oauth2/token',
+            headers=headers,
+            data=payload,
+            method="POST"
+        )
+        status_code = resp.code
+        if status_code == 200:
+            body = resp.read()
+        self.auth_token = json.loads(body)["access_token"]
+
+    def _query(self):
+        results = []
+        workloads = []
+        self._authenticate()
+        for stack_slug in self.stack_slugs:
+            try:
+                workloads = self._stackpath_query_get_list(self.api_host + '/workload/v1/stacks/' + stack_slug + '/workloads')
+            except Exception:
+                raise AnsibleError("Failed to get workloads from the StackPath API: %s" % traceback.format_exc())
+            for workload in workloads:
+                try:
+                    workload_instances = self._stackpath_query_get_list(
+                        self.api_host + '/workload/v1/stacks/' + stack_slug + '/workloads/' + workload["id"] + '/instances'
+                    )
+                except Exception:
+                    raise AnsibleError("Failed to get workload instances from the StackPath API: %s" % traceback.format_exc())
+                for instance in workload_instances:
+                    if instance["phase"] == "RUNNING":
+                        instance["stackSlug"] = stack_slug
+                        instance["workloadId"] = workload["id"]
+                        instance["workloadSlug"] = workload["slug"]
+                        instance["cityCode"] = instance["location"]["cityCode"]
+                        instance["countryCode"] = instance["location"]["countryCode"]
+                        instance["continent"] = instance["location"]["continent"]
+                        instance["target"] = instance["metadata"]["labels"]["workload.platform.stackpath.net/target-name"]
+                        try:
+                            ip = instance[self.hostname_key]
```

Comment:
```suggestion
                            ip = instance[self.hostname_key]
```
F841 local variable 'ip' is assigned to but never used